### PR TITLE
trim dependency tree: use os-lib only for tests

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

File: core/src/main/scala/replpp/util/package.scala [hunk 1]
```diff
@@ -14,5 +16,18 @@ package object util {
...
+
+  def linesStreamFromFile(path: Path): IterableOnce[String] =
+    Files.lines(path).iterator().asScala
+
+  def deleteRecursively(path: Path): Unit = {
+    if (Files.isDirectory(path))
+      Files.list(path).forEach(deleteRecursively)
+
+    Files.deleteIfExists(path)
+  }
```
Comment (maltek) [inline core/src/main/scala/replpp/util/package.scala:31]:
```suggestion
  def deleteRecursively(path: Path): Unit = {
    Files.walk(path, maxDepth = 0).forEach(Files.deleteIfExists)
  }
```

Comment (maltek) [reply core/src/main/scala/replpp/util/package.scala:31]: There's also the question of what do to with exceptions - do nothing like now, ignore exceptions, or rethrow but only after deleting everything that can be deleted. (But maybe it doesn't really matter with this only being used for scripts.)

**Review by maltek — APPROVED** [review]

File: core/src/main/scala/replpp/util/package.scala (hunk 1 above)
Comment (mpollmeier) [reply core/src/main/scala/replpp/util/package.scala:31]: your suggested change has different semantics - the idea is to recursively delete the given file or directory. your suggestion doesn't go deep (maxDepth=0) and fails for nonEmpty directories

Comment (maltek) [reply core/src/main/scala/replpp/util/package.scala:31]: Yes, I realized that a second after I pressed submit and ninja-edited the suggestion to drop the `maxDepth=0`. (Before that, I thought `maxDepth=0` would be no recursion limit. Then I checked again and realized it behaves like you say and there's an overload that doesn't require `maxDepth` at all.)

**Pull request merged**